pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ClientConfigurationData.java

```diff
@@ -46,9 +47,11 @@
 
     private String serviceUrl;
     @JsonIgnore
+    @EqualsAndHashCode.Exclude
```

Comment:
Same, the implementations of `ServiceUrlProvider` should ensure a meaningful "equals()` methods